src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -2555,23 +2556,16 @@ public async Task SendAsync_RequestVersionNotSpecified_ServerReceivesVersion11Re
             Assert.Equal(new Version(1, 1), receivedRequestVersion);
         }
 
-        [ActiveIssue(23037, TestPlatforms.AnyUnix)]
         [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Specifying Version(2,0) throws exception on netfx")]
         [OuterLoop("Uses external server")]
-        [Theory]
+        [ConditionalTheory]
         [MemberData(nameof(Http2Servers))]
         public async Task SendAsync_RequestVersion20_ResponseVersion20IfHttp2Supported(Uri server)
         {
             if (PlatformDetection.IsWindows && !PlatformDetection.IsWindows10Version1703OrGreater)
             {
                 // Skip this test if running on Windows but on a release prior to Windows 10 Creators Update.
-                _output.WriteLine("Skipping test due to Windows 10 version prior to Version 1703.");
-                return;
-            }
-            if (UseSocketsHttpHandler)
-            {
-                // TODO #23134: SocketsHttpHandler doesn't yet support HTTP/2.
-                return;
+                throw new SkipTestException("Skipping test due to Windows 10 version prior to Version 1703.");
```

Comment:
I didn't know we had `SkipTestException`.  It seems we should go through our existing tests and change all the simple `return` statements to use this form.

Reply:
This was added somewhat recently. It only works inside ConditionalTheory and ConditionalFact. 
If you throw this in "normal" fact, it will cause test failure.

Reply:
Ok.

But why are we skipping the test at all.  "Skipping test due to Windows 10 version prior to Version 1703."  That doesn't really explain why.  If the SocketsHttpHandler is our own HTTP/2 stack, why is there a dependency on a version of Windows 10?

Reply:
Perhaps we need to skip it if it is "IsWinHttpHandler" only?  That would be the only reason because WinHTTP didn't add support for HTTP/2  client-side until that version??

Reply:
This also runs as platform test. Is older winhttp able to send 2.0 request?

Reply:
Perhaps we need to skip it if it is "IsWinHttpHandler" only? That would be the only reason because WinHTTP didn't add support for HTTP/2 client-side until that version??

Reply:
>This also runs as platform test. Is older winhttp able to send 2.0 request?

Yes.  But that "skip" should be inside an "if (IsWinHttpHandler)"